src/editmap.cpp

```diff
@@ -1680,6 +1680,7 @@ int editmap::mapgen_preview( real_coords &tc, uimenu &gmenu )
                             memcpy( *destsm->frn, srcsm->frn, sizeof(srcsm->frn) ); // furniture
                             memcpy( *destsm->trp, srcsm->trp, sizeof(srcsm->trp) ); // traps
                             memcpy( *destsm->rad, srcsm->rad, sizeof(srcsm->rad) ); // radiation
+                            memcpy( *destsm->comp, srcsm->comp, sizeof(srcsm->comp) );
```

Comment:
This is wrong, the computer struct contains complex objects like the string and vectors that must be copied or moved through their respective operators. Copying the bytes will not work, bad things will happen, I predict crashes. You should swap this like the cosmetics map below.